### ISSUE TITLE
Multiple versions of the static html saved into the additionalhtmlhead setting

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-local_envbar#47]
If you edit via:

/admin/settings.php?section=additionalhtml

then the envbar doesn't bootstrap, and so the config element will not be resaved with a static copy of the envbar output.

However if you edit via this page:

https://moodle-upgrade.cqu.edu.au/admin/search.php?query=additionalhtml

This is will. So if you save it 3 times you will get 3 copies of the static html baked into that config item.

The root cause is the timing of the execution which we cannot do anything about. So what I think the most viable solution is a post save cleanup.

* [ ] in the renderer we need some unchanging tokens which we can use to detect the present and extend of the static html. I would suggest adding something like:

```
<!-- STARTENVBAR -->
<div class="envbar">...
<!--ENDENVBAR -->
```

* [ ] when the envbar bootstraps, it should quickly test $CFG->additionalhtml to see if it contains STARTENVBAR, and if so then it will regex out that chunk and resave the config item.

**Comment by brendanheywood** [comment]
looks good :+1:

**Comment by nhoobin** [comment]
Closing this off as it was implemented in pull request #48
